node_package/src/registerServerComponent.ts

```
@@ -0,0 +1,20 @@
+import React from 'react';
+import ReactOnRails from './ReactOnRails';
+import RSCClientRoot from './RSCClientRoot';
+import { RegisterServerComponentOptions } from './types';
+
+const registerServerComponent = (options: RegisterServerComponentOptions, ...componentNames: string[]) => {
+  const componentsWrappedInRSCClientRoot = componentNames.reduce(
+    (acc, name) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)